microsetta_public_api/api/plotting.py

```diff
@@ -1,11 +1,76 @@
-def plot_alpha_filtered(alpha_metric, age_cat=None, bmi_cat=None,
-                        percentiles=None, sample_id=None):
-    raise NotImplementedError()
+import pandas as pd
+import altair as alt
+from microsetta_public_api.repo._alpha_repo import AlphaRepo
+from microsetta_public_api.repo._metadata_repo import MetadataRepo
+from microsetta_public_api.models._alpha import Alpha
+from microsetta_public_api.api.metadata import _format_query, \
+    _validate_query, _filter_matching_ids
+from microsetta_public_api.utils._utils import jsonify
+
+
+def plot_alpha_filtered(alpha_metric=None, percentiles=None,
+                        sample_id=None, **kwargs):
+    repo = MetadataRepo()
+    query = _format_query(kwargs)
+    is_invalid = _validate_query(kwargs, repo)
+    if is_invalid:
+        return is_invalid
+
+    return _plot_alpha_percentiles_querybuilder(alpha_metric, percentiles,
+                                                query, repo, sample_id)
 
 
-def plot_alpha_filtered_json_query(body, alpha_metric, percentiles=None,
+def plot_alpha_filtered_json_query(body, alpha_metric=None, percentiles=None,
                                    sample_id=None):
-    raise NotImplementedError()
+    repo = MetadataRepo()
+
+    return _plot_alpha_percentiles_querybuilder(alpha_metric, percentiles,
+                                                body, repo, sample_id)
+
+
+def _plot_alpha_percentiles_querybuilder(alpha_metric, percentiles, query,
```

Comment:
I think the query matching may make sense to perform higher in the stack? The primary responsibility of this function seems like it's supposed to be plotting, but right now it's doing some filtering, validation of the filtering, and then plotting

Reply:
Either higher in the stack, or separate this function out into sub-functions. I think either option will make things more readable/organized.
Good catch, thanks!

Reply:
good call -- just a decomposition into a separate function would work too and may be better

Reply:
Done in https://github.com/biocore/microsetta-public-api/pull/41/commits/f6ef74820fd21ca9e603dffbd71a6ad4f5c11d08